### PR TITLE
Update comment

### LINKED ISSUE #54: Edit GitHub PR comments instead of posting a new one
As it stands, every time a PR-build that uses cimetrics is run, a new report/comment is posted on the GitHub PR. This is nice as it shows the historical evolution of the metrics since the PR was opened. However, it is very verbose.

Instead, we should edit the original comment with a new report for every additional build on the PR (instead of posting a new comment). See https://developer.github.com/v3/issues/comments/#edit-a-comment.

### PR DESCRIPTION
**Pull request by achamayou**
fix #54

### CODE REVIEW

**Comment by ghost** [comment]
update@[362 aka 20200110.15](https://dev.azure.com/jumaffre/metrics-devops/_build/results?buildId=362&view=results) vs master ewma over 40 builds from [129](https://dev.azure.com/jumaffre/metrics-devops/_build/results?buildId=129&view=results) to [346](https://dev.azure.com/jumaffre/metrics-devops/_build/results?buildId=346&view=results)
images

File: cimetrics/github_pr.py [hunk 1]
```diff
@@ -63,16 +64,40 @@ def upload_image(self, encoded_image):
         else:
             raise Exception("Failed to upload image")
 
+    def first_self_comment(self):
```
Comment (jumaffre) [inline cimetrics/github_pr.py:67]: Should this be the latest self comment instead?

**Comment by jumaffre** [comment]
I'd be keen to keep the old behaviour under a `metrics.yml` config parameter. Default would be to edit the latest comment but it should still be possible have a comment per build.

File: cimetrics/github_pr.py (hunk 1 above)
Comment (achamayou) [reply cimetrics/github_pr.py:67]: The implementation is the first, and in principle, we should only ever have one, so it should be the same? Is there a scenario/reason where's useful to take the last one instead?
All other things being equal, it's more work/execution/slower to get the last, so unless there's a use case for it, I prefer the first.

**Comment by achamayou** [comment]
@jumaffre sure, we can do that. Note that if you replace, you don't lose old ones, because the header for the comment "edited etc" lets you look at all comment revisions.

With this and the trend view, I'd rather take a side before 1.0 to simplify things (one thing I have in mind is we're going to want to build tests at some point). I think having a switch is fine for now, but I would like to converge on an opinion before we commit to support too many combinations of settings.

**Comment by jumaffre** [comment]
> @jumaffre sure, we can do that. Note that if you replace, you don't lose old ones, because the header for the comment "edited etc" lets you look at all comment revisions.
> 
> With this and the trend view, I'd rather take a side before 1.0 to simplify things (one thing I have in mind is we're going to want to build tests at some point). I think having a switch is fine for now, but I would like to converge on an opinion before we commit to support too many combinations of settings.

I think you're right. You can still see the history of comments in the "edited" menu. I'm happy to remove the old behaviour altogether.

File: cimetrics/github_pr.py (hunk 1 above)
Comment (jumaffre) [reply cimetrics/github_pr.py:67]: I was thinking of the case of keeping both the old (one post per build) and new (edit one unique comment) behaviours. If the PR was started with the old behaviour and say, 5 builds were run, then the metrics file updated to the new behaviour, then only the oldest report (out of 5) would be updated.

If we are happy to remove the old behaviour (I am), then this is a not an issue,

**Comment by achamayou** [comment]
Ok, let's go with this for now then.

**Pull request merged**